core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -55,29 +98,201 @@ func (l *logPollerWrapper) Ready() error {
-		err := subscriber.UpdateRoutes(l.routerContract, l.routerContract)
+		err := subscriber.UpdateRoutes(l.routerContract.Address(), l.routerContract.Address())
 		l.lggr.Errorw("LogPollerWrapper: Failed to update routes", "subscriberName", subscriberName, "error", err)
-	} else if l.contractVersion == 1 {
+	} else if l.pluginConfig.ContractVersion == 1 {
 		l.mu.Lock()
 		defer l.mu.Unlock()
 		l.subscribers[subscriberName] = subscriber
 
-		// TODO remove when periodic updates are ready
-		_ = subscriber.UpdateRoutes(l.routerContract, l.routerContract)
+		// TODO remove when periodic updates are ready and integration tests are properly migrated
+		_ = subscriber.UpdateRoutes(l.routerContract.Address(), l.routerContract.Address())
 	}
 }
+
+func (l *logPollerWrapper) checkForRouteUpdates() {
+	defer l.closeWait.Done()
+	freqSec := l.pluginConfig.ContractUpdateCheckFrequencySec
+	if freqSec == 0 {
+		l.lggr.Errorw("ContractUpdateCheckFrequencySec is zero - route update checks disabled")
+		return
+	}
+
+	updateOnce := func() {
+		// NOTE: timeout == frequency here, could be changed to a separate config value
+		timeoutCtx, cancel := utils.ContextFromChanWithTimeout(l.stopCh, time.Duration(l.pluginConfig.ContractUpdateCheckFrequencySec)*time.Second)
+		defer cancel()
+		active, proposed, err := l.getCurrentCoordinators(timeoutCtx)
+		if err != nil {
+			l.lggr.Errorw("LogPollerWrapper: error calling getCurrentCoordinators", "err", err)
+		}
+		l.handleRouteUpdate(active, proposed)
+	}
+
+	updateOnce() // update once right away
+	ticker := time.NewTicker(time.Duration(freqSec) * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-l.stopCh:
+			return
+		case <-ticker.C:
+			updateOnce()
+		}
+	}
+}
+
+func (l *logPollerWrapper) getCurrentCoordinators(ctx context.Context) (common.Address, common.Address, error) {
+	if l.pluginConfig.ContractVersion == 0 {
+		return l.routerContract.Address(), l.routerContract.Address(), nil
+	}
+	var donId [32]byte
+	copy(donId[:], []byte(l.pluginConfig.DONId))
+
+	activeCoordinator, err := l.routerContract.GetContractById(&bind.CallOpts{
+		Pending: false,
+		Context: ctx,
+	}, donId, false)
+	if err != nil {
+		return common.Address{}, common.Address{}, err
+	}
+
+	proposedCoordinator, err := l.routerContract.GetContractById(&bind.CallOpts{
+		Pending: false,
+		Context: ctx,
+	}, donId, true)
+	if err != nil {
+		return common.Address{}, common.Address{}, err
+	}
+
+	return activeCoordinator, proposedCoordinator, nil
+}
+
+func (l *logPollerWrapper) handleRouteUpdate(activeCoordinator common.Address, proposedCoordinator common.Address) {
+	changed := false
+	l.mu.Lock()
+	if activeCoordinator != l.activeCoordinator || proposedCoordinator != l.proposedCoordinator {
+		l.activeCoordinator = activeCoordinator
+		l.proposedCoordinator = proposedCoordinator
+		changed = true
+	}
+	l.mu.Unlock()
+	if !changed {
+		l.lggr.Debug("LogPollerWrapper: no changes to routes")
```

Comment:
Should this be a `trace` log since this is going to be logged so frequently?

Reply:
Only once every pluginConfig.ContractUpdateCheckFrequencySec seconds. Debug is fine.